testing/upload_validation.ts

```diff
@@ -1918,7 +1924,9 @@ ${endpointKey ? 'endpointKey is set' : `requiresEndpointUrl is ${requiresEndpoin
     // TODO(jonathan): See if we should really allow this. The SDK right now explicitly tolerates an undefined
     // schema for objects, but that doesn't seem like a use case we actually want to support.
     schema: z.union([genericObjectSchema, arrayPropertySchema]).optional(),
-  });
+  }).refine(val => {
+    return !val.validateParameters;
+  }, {message: 'validateParameters is only allowed on sync formulas.'});
 
   const formulaMetadataSchema = z
```

Comment:
Can you just do the check here instead of in the separate union members?

Reply:
Yes done